docs/readmes/proposals/p005_call_tracing.md

```diff
@@ -0,0 +1,283 @@
+subscriber, protocol, or interface.
+
+## Phase 1
+
+#### User Flow
+
+For call tracing with Magma, we will provide the following user flow:
+1. User accesses NMS
+2. User selects either the gateway or subscriber that they would like to do
+call tracing for
+3. User starts the call trace
+4. After a timeout, or by user control, the call trace stops
+5. The call trace will be provided as a download to the user
+
+### Design
+
+#### Orchestrator - API
+
+The API supports starting the call trace, ending the tracing session, and
+downloading the resulting capture.
+
+The API will be as follows:
+```
+GET      /networks/{network_id}/call_tracing/{trace_id}
+POST     /networks/{network_id}/call_tracing/{trace_id}
+PUT      /networks/{network_id}/call_tracing/{trace_id}
+DELETE   /networks/{network_id}/call_tracing/{trace_id}
+
+GET      /networks/{network_id}/call_tracing/download/{trace_id}
+```
+
+To use call tracing via the REST API, the following steps should be followed.
+
+To begin a call trace, the POST endpoint should be called to create a
+`call_trace` resource. During this time, the type of call trace should be
+specified, along with timeout, and the `trace_id`.
+
+To end a call trace, either the user can wait for the timeout, or to manually
+end the call trace, the PUT endpoint can be called, specifying `requested_end`
+as `true`.
+
+To download the call trace, the GET endpoint should return whether the call
+trace has ended and is ready to download, and if it is, will return the
+download URL.
+
+```
+definitions:
+  call_trace:
+   description: Mutable Call Trace
+   type: object
+   required:
+     - trace_id
+     - config
+   properties:
+     trace_id:
+       type: string
+       x-nullable: false
+       example: "pair1"
+     config:
+       $ref: call_trace_config
+     state:
+       $ref: call_trace_state
+
+  mutable_call_trace:
+   description: Mutable Call Trace
+   type: object
+   required:
+     - trace_id
+     - config
+   properties:
+     trace_id:
+       type: string
+       x-nullable: false
+       example: "pair1"
+     requested_end:
+       type: boolean
+     config:
+       $ref: call_trace_config
+
+  call_trace_config:
+    type: object
+    description: Call Trace spec
+    required:
+      - type
+    properties:
+      trace_type:
+        type: string
+        x-nullable: false
+        enum:
+          - 'GATEWAY'
+          - 'SUBSCRIBER'
+          - 'PROTOCOL'
+          - 'INTERFACE'
+      imsi:
+        type: string
+      protocol:
+        type: string
+      interface:
+        type: string
+      timeout:
+        type: integer
+        format: uint32
+        description: Timeout of call trace in seconds
+
+  call_trace state:
+    type: object
+    description: Full state object of a call trace
+    properties:
+      call_trace_available:
+        type: boolean
+```
+
+#### Orchestrator - ctraced
+
+On the gateway, the ctraced service will manage call tracing for the network.
```

Comment:
To some extent, if we make this service generic, we could potentially use it for port mirroring features on behalf of the user. Though that is not the goal here. This might be in case user wants to capture traffic on actual physical interfaces or if we support capturing traffic samples on the port.

Reply:
Should I leave this section as-is?

Reply:
Yes. I think that sounds fine.